packages/mui-material/src/Backdrop/Backdrop.test.js

```diff
@@ -15,18 +15,16 @@ describe('<Backdrop />', () => {
     refInstanceof: window.HTMLDivElement,
     muiName: 'MuiBackdrop',
     testVariantProps: { invisible: true },
-    testLegacyComponentsProp: true,
     slots: {
-      root: {
-        expectedClassName: classes.root,
+      transition: {
```

Comment:
yeah, I mentioned it [here](https://github.com/mui/material-ui/pull/40677/commits/eb67e729e9007f458f652319f7ddf1190d5ba873#r1481720438), and was unsure what to do with it.

Reply:
I see, sorry for missing that. This is due to outdated proptypes. The [root slot proptype](https://github.com/mui/material-ui/blob/6f8ff5cd073963530675ba8cd81f0e48e5a34f8a/packages/mui-material/src/Backdrop/Backdrop.js#L163) should be:

```diff
-root: PropTypes.object,
+root: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
``` 

Running `yarn proptypes` should change that proptype to the correct one (or `pnpm proptypes`, we recently changed so if you rebased recently it might be `pnpm` instead of `yarn`).

If it doesn't, try deleting the proptype and running `yarn proptypes` again.

Reply:
pnpm seems really insistent that root should be of type "PropTypes.object", and unfortunallty not what you suggested. I've tried three times now, once with just running pnpm proptype, and twice deleting the proptypes and running pnpm proptype. 

Do you have an idea what exactly could be causing this?

Reply:
Good news! figured it out... The root SlotsAndSlotProps was not wrapped in the SlotProps. 🙃 Or at least I think this was the problem, let me know your thoughts.

![Screenshot 2024-02-16 at 10 05 15](https://github.com/mui/material-ui/assets/79278353/6477b0b0-428b-4baa-9d7f-99bf23d7f21a)

Reply:
Nice! Yes, that seems like the problem 🙌🏼

Reply:
I've got one last error before it can get reviewed... I'll investigate it, but thought it would just be worth a quick mention incase you've seen it before.

![Screenshot 2024-02-16 at 14 37 51](https://github.com/mui/material-ui/assets/79278353/4c2619b2-134b-465e-91e0-d412a5958a47)

Reply:
We'll need to use the `useSlot` hook for the root as well, so:

``` tsx
// packages/mui-material/src/Backdrop/Backdrop.js

// ...

  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };

  const externalForwardedProps: {
    slots: backwardCompatibleSlots,
    slotProps,
  }

  const [RootSlot, rootProps] = useSlot('root', {
    elementType: BackdropRoot,
    externalForwardedProps,
    className: clsx(classes.root, className),
    ownerState,
  });

  const [TransitionSlot, transitionProps] = useSlot('transition', {
    elementType: Fade,
    externalForwardedProps,
    ownerState,
  });

  delete transitionProps.ownerState;

  return (
    <TransitionSlot in={open} timeout={transitionDuration} {...other} {...transitionProps}>
      <RootSlot
        aria-hidden
        {...rootProps}
        classes={classes}
        ref={ref}
      >
        {children}
      </RootSlot>
    </TransitionSlot>
  );

// ...
```

I think that should do it

Reply:
and we have lift off, it worked! Thanks